include/libtorrent/torrent.hpp

```diff
@@ -401,6 +401,11 @@ namespace libtorrent {
 		// before disconnecting
 		bool graceful_pause() const { return m_graceful_pause_mode; }
 
+		boost::uint64_t flags() const;
+		void set_flags(boost::uint64_t flags, boost::uint64_t mask);
+		void set_flags(boost::uint64_t flags) { set_flags(0xffffffffffffffff, flags); }
+		void unset_flags(boost::uint64_t flags) { set_flags(0, flags); }
```

Comment:
I think these two convenience functions should just exist on the user-facing ``torrent_handle``. I think that would simplify things.